### PR TITLE
Chore: Initializing notifications in viewer rather than annotator

### PR DESCRIPTION
**Pull request by pramodsum**
- This will allow us to display other types of notifications should we have the need

### CODE REVIEW

**Comment by pramodsum** [comment]
@JustinHoldstock So I thought about this and I think that the notification should be kept inside PreviewUI.js rather than Preview.js

This does mean that we will instantiate a notification on a per viewer basis still but it seems to make the most sense for it to live there with other UI components such as the progress bar, etc

**Comment by JustinHoldstock** [comment]
@pramodsum Are we creating a new instance of the Loading bar and spinner for every viewer?

File: src/lib/PreviewUI.js [hunk 1]
```diff
@@ -285,6 +289,28 @@ class PreviewUI {
         this.progressBar.finish();
     }
 
+    /**
+     * Shows a notification message.
+     *
+     * @private
```
Comment (JustinHoldstock) [inline src/lib/PreviewUI.js]: Should be public, as it's being called in Preview.js (and needs to be called by anything else using it)

Ditto to `hideNotification()`

File: src/lib/annotations/Annotator.js [hunk 2]
```diff
@@ -725,9 +718,29 @@ class Annotator extends EventEmitter {
...
+     * @protected
+     * @emits viewerevent
+     * @param {string} event - Event name
+     * @param {Object} data - Event data
+     * @return {void}
+     */
+    emit(event, data) {
+        const { annotator, fileId } = this.options;
+        super.emit(event, data);
+        super.emit('annotationevent', {
```
Comment (JustinHoldstock) [inline src/lib/annotations/Annotator.js]: What's the reason we're emitting double events?

**Comment by pramodsum** [comment]
Actually I look another look. Seems like we're instantiating one instance of PreviewUI in Preview.js so that should apply for all viewers.

File: src/lib/annotations/image/ImageAnnotator.js [hunk 3]
```diff
@@ -105,7 +105,7 @@ class ImageAnnotator extends Annotator {
             type
         };
 
-        if (!annotatorUtil.validateThreadParams(threadParams)) {
+        if (annotatorUtil.validateThreadParams(threadParams)) {
```
Comment (JustinHoldstock) [inline src/lib/annotations/image/ImageAnnotator.js]: Does this mean that if `validateThreadParams()` returns `true`, that it isn't valid?

Comment (pramodsum) [inline src/lib/annotations/image/ImageAnnotator.js]: Ah! I was forcing this to trigger the emission of annotation errors. Will fix that up

**Pull request merged**